datadog/resource_datadog_dashboard.go

```diff
@@ -4722,6 +4722,257 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
+								Description: "Number of results to return",
+							},
+							"order": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "desc",
+								Description: "Direction of sort.",
+							},
+						},
+					},
+				},
+				"alias": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Expression alias.",
+				},
+			},
+		},
+	}
+}
+
+func getFormulaQuerySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"metric_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions metrics query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "metrics",
+								Description: "Data source for metrics queries.",
+							},
+							"query": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Metrics query definition.",
+							},
+							"aggregator": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "The aggregation methods available for metrics queries.",
+							},
+							"name": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Name of the query for use in formulas.",
+							},
+						},
+					},
+				},
+				"event_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions events query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Data source for event platform-based queries.",
+							},
+							"search": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								MaxItems:    1,
+								Description: "Search options.",
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"query": {
+											Type:         schema.TypeString,
+											ValidateFunc: validators.ValidateStringValue,
+											Required:     true,
+											Description:  "Events search string.",
+										},
+									},
+								},
+							},
+							"indexes": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+								Description: "An array of index names to query in the stream.",
+							},
+							"compute": {
+								Type:        schema.TypeList,
+								Required:    true,
+								Description: "Compute options.",
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"aggregation": {
+											Type:        schema.TypeString,
```

Comment:
Can validate against an enum as well.

Reply:
Done in [commit 2f914c9](https://github.com/DataDog/terraform-provider-datadog/pull/892/commits/2f914c9da8e833c9a46c5c24bd654aa47ba44918)